libs/libc/string/lib_strsignal.c

```diff
@@ -96,80 +69,51 @@ FAR char *strsignal(int signum)
 
   switch (signum)
     {
+#ifdef CONFIG_LIBC_STRSIGNAL
```

Comment:
what's difference between CONFIG_LIBC_STRSIGNAL=y/CONFIG_LIBC_STRSIGNAL_SHORT=n and CONFIG_LIBC_STRSIGNAL=n?

Reply:
> CONFIG_LIBC_STRSIGNAL=y/CONFIG_LIBC_STRSIGNAL_SHORT=n and CONFIG_LIBC_STRSIGNAL=n

`CONFIG_LIBC_STRSIGNAL=y` enables `strsignal()` to produce strings different from `"Signal X"` for well known signal numbers.
`CONFIG_LIBC_STRSIGNAL_SHORT=n` will enable more human friendly strings compared to `y` when the string will be just a string alias `SIGHUP` will get `"SIGHUP"` string. 
Please take a look into the Kconfig description.

Reply:
why not follow strerror? Implmenting strsignal alwaysl

Reply:
It follows the same approach. The strsignal() is always implemented as strerror() is as well.

Reply:
@xiaoxiang781216 please resolve if no more questions here